libraries/AP_Math/quaternion.cpp

```diff
@@ -327,6 +331,23 @@ Quaternion Quaternion::operator*(const Quaternion &v) const
     return ret;
 }
 
+Quaternion Quaternion::operator*(const float &k) const
+{
+	Quaternion ret;
```

Comment:
your use of mixed whitespace is quite crazy.

Reply:
do you prefer me to use only spaces instead of tabs?

Reply:
Bring consistent is a good start. You have other whitespace issues but as a new contributor we are happy to clean it up for you later. So, don't worry about it.

For next time, please take a look at 
http://ardupilot.org/dev/docs/contributing.html

General rule is, make your code look just like all the other million lines of code around it. You have lots of examples to go from. Mixed styles make it hard to read at a glance.